src/main/java/com/money_account_service/services/AccountService.java

```diff
@@ -2,30 +2,56 @@
 
 
 import com.money_account_service.dtos.request.CreateAccountRequestDto;
+import com.money_account_service.dtos.response.AccountDetailsResponseDto;
 import com.money_account_service.dtos.response.CreateAccountResponseDto;
+import com.money_account_service.entities.AccountEntity;
+import com.money_account_service.mappers.ResponseMapper;
 import com.money_account_service.models.UserModel;
 import com.money_account_service.repositories.AccountRepository;
-import com.money_account_service.utility.UserServiceClient;
+import com.money_account_service.utility.DateTimeProvider;
 import lombok.AllArgsConstructor;
+import org.iban4j.CountryCode;
+import org.iban4j.Iban;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.Optional;
 
 @AllArgsConstructor
 public class AccountService {
 
     private AccountRepository accountRepository;
 
+    private DateTimeProvider dateTimeProvider;
+
     public CreateAccountResponseDto createAccount(CreateAccountRequestDto createAccountRequestDto, UserModel userModel) {
-//        if (authorizeRequest(createAccountRequestDto.accessToken()).authorized()) {
-////            AccountEntity accountEntity = accountRepository.save(RequestMapper.accountRequestToAccountEntity(createAccountRequestDto));
-////            return ResponseMapper.accountEntityToCreateAccountResponse(a)
-//            return null;
-//        }
-        System.out.println(userModel);
-        return null;
+        AccountEntity accountEntity = AccountEntity.builder()
+                .accountNumber(generateAccountNumberForUser())
+                .currency(createAccountRequestDto.currency())
+                .userSub(userModel.userSub())
+                .createdAt(dateTimeProvider.getInstant())
+                .updatedAt(dateTimeProvider.getInstant())
+                .build();
+
+        if (accountRepository.findByUserSub(accountEntity.getUserSub()).isPresent())
```

Comment:
You have quite often this structure `repository.findBy(...).orThrow(...)`. I think we could add such method to the repository and throw some kind of `NotFoundException`. Later, we could have exception mapper to the response, e.g. like this https://www.baeldung.com/exception-handling-for-rest-with-spring.